appointment/src/main/java/com/bmwapp/appointment/controller/AppointmentController.java

```diff
@@ -50,30 +54,30 @@ public AppointmentsResponse getAllAppointments(@RequestParam(value = "customerId
                 .collect(Collectors.toList()));
     }
 
+    @PreAuthorize("hasAuthority('CUSTOMER')")
     @GetMapping("/forCustomer")
     public AppointmentsResponse getAllAppointmentsForCustomer() {
-        contextProvider.requiredAnyRoles("CUSTOMER", "AGENT");
         log.info("Get all customers appointments");
         List<Appointment> appointments = appointmentService.getAllAppointmentsByCustomerId(contextProvider.getLoggedUserId());
         return new AppointmentsResponse(appointments.stream()
                 .map(appointmentService::convertToDto)
                 .collect(Collectors.toList()));
     }
 
+    @PreAuthorize("hasAnyAuthority('CUSTOMER', 'AGENT')")
     @PutMapping
     public AppointmentResponse updateAppointment(@RequestBody AppointmentUpdateRequest appointmentUpdateRequest) {
-        contextProvider.requiredAnyRoles("CUSTOMER", "AGENT");
         log.info("Update appointment {}", appointmentUpdateRequest);
         Appointment appointment = appointmentService.convertToEntity(appointmentUpdateRequest.appointmentDto());
         Appointment updatedAppointment = appointmentService.updateAppointment(appointment);
         return new AppointmentResponse(appointmentService.convertToDto(updatedAppointment));
     }
 
+    @PreAuthorize("hasAuthority('CUSTOMER')")
```

Comment:
for me it should be avaiailable for both customer and agent

Reply:
(agent should be able to remove all appointments) - it can be a separate endpoint